src/ciflow-bot.ts

```diff
@@ -0,0 +1,220 @@
+import * as probot from 'probot';
+
+export class CIFlowBot {
+  // Constructor required
+  readonly ctx: probot.Context;
+
+  // Static readonly configurations
+  static readonly allowed_commands: string[] = ['ciflow'];
+  static readonly bot_app_name = 'pytorchbot';
+  static readonly bot_assignee = 'pytorchbot';
+  static readonly event_issue_comment = 'issue_comment';
+  static readonly event_pull_request = 'pull_request';
+  static readonly pr_label_prefix = 'ciflow/';
+  static readonly rollout_users = ['zhouzhuojie']; // slow rollout to specific group of users first
+  static readonly strategy_add_default_labels = 'strategy_add_default_labels';
+
+  // Stateful instance variables
+  command = '';
+  command_args: string[] = [];
+  comment_author = '';
+  comment_body = '';
+  dispatch_labels: string[] = [];
+  dispatch_strategies = [CIFlowBot.strategy_add_default_labels];
+  event = '';
+  owner = '';
+  pr_author = '';
+  pr_labels: string[] = [];
+  pr_number = 0;
+  repo = '';
+
+  constructor(ctx: probot.Context) {
+    this.ctx = ctx;
+  }
+
+  valid(): boolean {
+    if (
+      this.event !== CIFlowBot.event_pull_request &&
+      this.event !== CIFlowBot.event_issue_comment
+    ) {
+      this.ctx.log.error({ctx: this.ctx}, 'Unknown webhook event');
```

Comment:
This is, like, an internal assert, because these are the only events we subscribed to, so if we get other events, something has gone seriously wrong, right? The error message doesn't convey the severity of this situation

Reply:
since I will be monitoring the logs once deployed, it helps for detection in the future if we extend the bot to listen on more events without proper handlers. We can throw an exception here, but it's the same thing from logging and alerting perspective.